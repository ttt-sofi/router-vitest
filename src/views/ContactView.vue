<template>
    <div class="contact">
      <h1>Vista de Contacto</h1>
      <p>Esta es la vista de contacto</p>
    </div>
  </template>
  
  <script>
  export default {
    name: "ContactView",
  };
  </script>