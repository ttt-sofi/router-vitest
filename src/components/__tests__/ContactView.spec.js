import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import ContactView from "@/views/ContactView.vue";
import {describe,test, expect} from "vitest";

describe("ContactView", () => {
    test("Probando la existencia del componente o vista ContactView ", async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [
                {
                    path: "/contact",
                    name: "ContactViewVue",
                    component: ContactView,
                },
            ],
        });
        router.push("/contact");
        await router.isReady();
        const wrapper = mount(ContactView, {
            global: {
                plugins: [router],
            },
        });
        expect(wrapper.findComponent(ContactView).exists()).toBe(true);
        expect(wrapper.html()).toMatchSnapshot();
        });

        test("DeberÃ­a contener el texto 'Vista de Contacto'", () => {
            const wrapper = mount(ContactView);
            expect(wrapper.text()).toContain("Vista de Contacto");
          });
});