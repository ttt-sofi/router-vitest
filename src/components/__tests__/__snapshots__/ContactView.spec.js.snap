// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactView > Probando la existencia del componente o vista ContactView  1`] = `
"<div class="contact">
  <h1>Vista de Contacto</h1>
  <p>Esta es la vista de contacto</p>
</div>"
`;
